### DISCUSSION TITLE
how do I send messages from the code-server's webview to the host page？code-server embed in an iframe

### DISCUSSION THREAD

**Discussion by alanhe421** [coder/code-server#6575] [General]
Assuming I have included code-server in an iframe, how do I send messages from the code-server's webview to the host page, that is, the webpage itself that contains the code-server within an iframe?

code-server: v4.14.0

**Comment by code-asher** [comment]
There is no way out of the box to do this, it would require new code.

For visibility, I will cross-post what I wrote here: https://github.com/coder/code-server/discussions/5205#discussioncomment-7833809

Making an extension communicate with the parent page is probably going to involve significant patching to VS Code, but it should be possible.

Vaguely implementation probably looks something like:

1. Add a listener for messages between the parent and child frames
2. Add or hijack a channel between the main thread and the extension host (as a random example see `mainThreadCommands.ts` and `extHostCommands.ts`)
3. Add API for extensions to use the new channel
